src/main/python/solr_integration_test.py

```diff
@@ -0,0 +1,210 @@
+        #### we check whether we have correct solr path first
+        solr_on = "solrini/bin/solr restart -c -m 8G"
+        solr_config = "pushd src/main/resources/solr && ./solr.sh ../../../../solrini localhost:9983 && popd"
+        self.command_execution("solrini/bin/solr", solr_on)
+        self.command_execution("src/main/resources/solr", solr_config)  
+        print("Solr server is on")
+
+
+    def turn_off_server(self):
+        subprocess.run("solrini/bin/solr stop -all",shell=True)        
+
+    # we consider 9983 as default local port number
+    # if this collection already exists on solr and has correct number of indexed document from previous test,we ask user 
+    # whether overwrite it, final MAP, P_30 result will be different for some reason. Thus we delete it and re-build it   
+    def indexing(self,name,input_path,thread_num):
+        thread_num = str(thread_num)
+        # if this collection already exists
+        collection_location = "solrini/server/solr/" + name + "_shard1_replica_n1"  
+        # if check_indexing return False, it mean previous indexed doc is incorrect,then we should overwrite this collection
+        # re-execute indexing, searching step.
+        if (os.path.exists(collection_location) and self.check_indexing() ):
+            response = input("collection already exists on Solr server and indexed doc number is same as expectancy\n \
+                    do you want to overwrite it? input [Y]es or [N]o \n")
+
+            if (response == "N"): 
+                print("We do not overwrite the existing collection,directly jump to Searching step")
+                return
+            print("We overwrite the existing collection {}  now".format(name))  
+            command0 = "solrini/bin/solr delete -c {}".format(name)
+            self.command_execution("solrini/bin/solr", command0)  
+
+        print("Create data collection on Solr")
+        command1 = "solrini/bin/solr create -n anserini -c {}".format(name)     
+        self.command_execution("solrini/bin/solr", command1)
+
+        collection, generator = index_options[name]["collection"],index_options[name]["generator"]
+        print("\nsolr indexing start for {}, {}, {}, {}, {} \n".format(collection,generator,name,input_path,thread_num))    
+        command2 = "sh target/appassembler/bin/IndexCollection -collection {}  -generator {} -threads {} -input {} \
+            -solr -solr.index {} -solr.zkUrl localhost:9983  -storePositions -storeDocvectors -storeTransformedDocs \
+            ".format(collection,generator,thread_num,input_path,name)
+        self.command_execution("target/appassembler/bin/IndexCollection", command2)   
+        ###subprocess.run(command2, shell = True)  
+
+    # if data collection already exists on solr server, check whether it has correct number of indexed document
+    # return true/false
+    def check_indexing(self):
+        subprocess.run("wget http://localhost:8983/solr/core18/query?q=*:*",shell=True)
```

Comment:
don't use `wget` in a process... use something like `requests`: https://2.python-requests.org//en/latest/